boa_engine/src/syntax/ast/node/declaration/mod.rs

```diff
@@ -276,6 +276,114 @@ impl DeclarationPattern {
             DeclarationPattern::Array(pattern) => pattern.init(),
         }
     }
+
+    /// Returns true if the node contains a identifier reference named 'arguments'.
+    ///
+    /// More information:
+    ///  - [ECMAScript specification][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-static-semantics-containsarguments
+    #[inline]
+    pub(crate) fn contains_arguments(&self) -> bool {
```

Comment:
Would be good to make this its own trait, akin to `ToInternedString`, to delegate the work to each node type individually

Reply:
Ah, maybe renaming `ToInternedString` to `AstNode` and creating a default implementation for `contains_arguments` would be better

Reply:
I though about this too. We could do it, but I think spreading these functions over all ast node files would probably make it unnecessarily hard to read. Especially since it it very trivial functionality. Also the spec defines this as one function, so it is easier to compare to that.
What do you think?

Reply:
Hm, yeah, it would be very unreadable to have parts of the same function through all the module